dask_planner/src/expression.rs

```diff
@@ -281,9 +305,44 @@ impl PyExpr {
                 operands.push(py_ex);
                 Ok(operands)
             }
-            _ => Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(
-                "unknown Expr type encountered",
-            )),
+            Expr::Case {
```

Comment:
We should be able to delegate to DataFusion to get the Arrow type for the expression, and then just map that to the dask type.

Reply:
Actually, never mind. I misunderstood what was happening here and thought this was determining the data type for an expression.

Reply:
No worries